latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/TemplateProcessor.java

```diff
@@ -159,4 +165,38 @@ public String render(String templateName) throws Exception
     {
         return render(getTemplate(templateName));
     }
+
+    private String getTemplateName(Block block)
+    {
+        // If there's a custom template defined in the Block parameter's, use it!
+        String templateName = block.getParameter("latex-template");
+        if (templateName == null) {
+            templateName = block.getClass().getSimpleName();
+        }
+        return templateName;
+    }
+
+    private Optional<String> renderUIXPs(String templateName, String suffix)
+    {
+        // The XDOM template case is particular, the after UIXP needs to be located before "\end{document}" which closes
+        // the latex document. To do so, the UIXP of XDOM are directly integrated in the template in Velocity.
+        final Optional<String> ret;
+        if (!templateName.equals("XDOM")) {
+            List<UIExtension> extensions =
+                this.uiExtensionSupplier.getExtensions(
+                    String.format("org.xwiki.contrib.latex.%s.%s", templateName, suffix));
+            if (extensions.isEmpty()) {
+                ret = Optional.empty();
+            } else {
+                DefaultWikiPrinter printer = new DefaultWikiPrinter();
+                for (UIExtension extension : extensions) {
```

Comment:
Do we want to enforce some sort of order on the UIXs? For instance using a `priority` parameter.

Reply:
I think it would be a good idea yes. It's quite easy to do so I think we should do it.